NitroxPatcher/Patches/Dynamic/LiveMixin_AddHealth_Patch.cs

```diff
@@ -45,19 +51,61 @@ public static void Postfix(float __state, LiveMixin __instance, bool __runOrigin
         }
     }
 
-    private static bool HandleRadiationLeakRepair(LiveMixin liveMixin)
+    private static bool ShouldBroadcastHeal(LiveMixin victim, bool processingPacket)
     {
-        if (!liveMixin.TryGetComponent(out RadiationLeak radiationLeak) || !liveMixin.TryGetNitroxId(out NitroxId leakId))
+        foreach (MonoBehaviour monoBehaviour in victim.GetComponents<MonoBehaviour>())
         {
-            return false;
+            switch (monoBehaviour)
+            {
+                case RadiationLeak radiationLeak:
+                    if (!processingPacket)
+                    {
+                        HandleRadiationLeakRepair(radiationLeak);
+                    }
+                    return false;
+                case BaseCell baseCell:
+                    if (!processingPacket)
+                    {
+                        HandleBaseLeakRepair(baseCell, victim);
+                    }
+                    return false;
+            }
+        }
+
+        return !processingPacket;
+    }
```

Comment:
This logic seems to do nothing when `processingPacket` is false.

I recommend changing this to return early, or better yet, not call this function so it doesn't need the extra argument.

In addition, `Should*` prefixed methods should not have side effects like sending packets. Rename the method to something like `BroadcastHeal` or `HandleBroadcastHeal`.

Reply:
Would this make sense?
```cs
private static bool TryBroadcastHeal(LiveMixin victim)
{
    foreach (MonoBehaviour monoBehaviour in victim.GetComponents<MonoBehaviour>())
    {
        switch (monoBehaviour)
        {
            case RadiationLeak radiationLeak:
                HandleRadiationLeakRepair(radiationLeak);
                return true;
            case BaseCell baseCell:
                HandleBaseLeakRepair(baseCell, victim);
                return true;
        }
    }
    return false;
}
```
and
```cs
public static void Postfix(float __state, LiveMixin __instance, bool __runOriginal)
{
    bool healthChanged = __state != __instance.health;
    if (!__runOriginal || !healthChanged)
    {
        return;
    }
    if (!Resolve<LiveMixinManager>().ProcessingRemoteHealthChange || TryBroadcastHeal(__instance))
    {
        return;
    }
    
    // Other code
```

Reply:
No we can't. We need to trigger either of the Handle functions because they contain some important code to be executed (it's not all broadcast code). This is why I put the "processingremotehealthchange" check later

Reply:
Yea I messed up the boolean logic, should have been:
`if (Resolve<LiveMixinManager>().ProcessingRemoteHealthChange && TryBroadcastHeal(__instance))`

If processing, try broadcast heal and if it was handled do nothing more. If not processing, continue on.

Reply:
But then you're missing the default case in which the object being healed is not a specific instance (so TryBroadcastHeal would return false in your situation) but still we're processing a packet (first condition returns true) then we're not returning while we should be.

Reply:
Not sure I get which default case you mean `switch` in `ShouldBroadcastHeal` has none.

If with default case you mean this code at the end of `Postfix`:
```cs
// Let others know if we have a lock on this entity
if (__instance.TryGetIdOrWarn(out NitroxId id) && Resolve<SimulationOwnership>().HasAnyLockType(id))
{
    Optional<EntityMetadata> metadata = Resolve<EntityMetadataManager>().Extract(__instance.gameObject);

    if (metadata.HasValue)
    {
        Resolve<Entities>().BroadcastMetadataUpdate(id, metadata.Value);
    }
}
```

Then it should still be called when either:
 - `Resolve<LiveMixinManager>().ProcessingRemoteHealthChange` is `false` (same as before, `ShouldBroadcastHeal` did nothing)
 - `ShouldBroadcastHeal(__instance, processingPacket)` failed to handle it before. (same as before but I inverted the return values).

Reply:
Then I'll just mix everything in Postfix (the current last part of it is what I meant by "default case" and it'll now be handled by the "default" case itself in the switch)

Reply:
Sounds good, might be more verbose but it's hard to keep track otherwise as an outsider.